contract_split/models/contract.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Damien Crier - Foodles
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class Contract(models.Model):
+    _inherit = "contract.contract"
+
+    original_contract_ids = fields.Many2many(
+        comodel_name="contract.contract",
+        relation="contract_split_contract_rel",
+        column1="contract_id",
+        column2="split_contract_id",
+        readonly=True,
+    )
+
+    @api.model
+    def _get_contract_split_name(self, split_wizard):
+        return split_wizard.main_contract_id.name
+
+    @api.model
+    def _get_values_create_split_contract(self, split_wizard):
+        return {
+            "name": self._get_contract_split_name(split_wizard),
+            "partner_id": split_wizard.partner_id.id,
+            "invoice_partner_id": split_wizard.invoice_partner_id.id,
+            "original_contract_ids": [split_wizard.main_contract_id.id],
+            "line_recurrence": True,
+        }
```

Comment:
I don't really understand why thoses methods are defined on the contract.contract ?

Probably I miss something but I've the feeling that it get more interest on the wizard and would make code easier to read ?!